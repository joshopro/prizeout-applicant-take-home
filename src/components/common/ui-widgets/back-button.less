@import '../../../utils/styles/mixins.less';

button {

	&.back-button {
		@icon-color: var(--color-blue-900);

		&.button {
			height: 1.75rem;
			width: 1.75rem;
			border: unset;
			color: @icon-color;
			padding: 0;
			border-radius: .5rem;

			&:hover {
				background-color: var(--color-grey-100);
				border-color: var(--color-grey-100);
			}

			&:active {
				background-color: var(--color-grey-200);
				border-color: var(--color-grey-200);
			}

			&:focus {
				box-shadow: unset;
			}
		}
	}
}