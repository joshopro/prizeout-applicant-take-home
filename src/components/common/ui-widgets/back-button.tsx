import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../form-components';

import './back-button.less';

interface BackButtonProps {
    onClick: () => void;
    orientation?: string;
}

export const BackButton: React.FC<BackButtonProps> = ({ onClick, orientation }): React.ReactElement => {
    const icon = orientation === 'right' ? 'far fa-chevron-right' : 'far fa-chevron-left';
    const ariaLabel = orientation === 'right' ? 'Move forward' : 'Move back';

    return <Button ariaLabel={ariaLabel} icon={icon} onClick={onClick} className="back-button" />;
};

BackButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    orientation: PropTypes.string,
};
