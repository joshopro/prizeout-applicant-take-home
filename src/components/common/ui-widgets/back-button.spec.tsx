import { render, screen } from '../../../testing/test-utils';
import userEvent from '@testing-library/user-event';
import React from 'react';

import { BackButton } from './back-button';

describe('Test Back Button Component', () => {
    const onClick = jest.fn();

    test('Component matches snapshot', () => {
        const { asFragment } = render(<BackButton onClick={onClick} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Component orientation is correct', () => {
        const { container } = render(<BackButton orientation="right" onClick={onClick} />);

        expect(container.getElementsByClassName('far fa-chevron-right').length).toBe(1);
    });

    test('Component can call onClick', () => {
        render(<BackButton onClick={onClick} />);

        userEvent.click(screen.getByTestId('button'));
        userEvent.click(screen.getByTestId('button'));
        expect(onClick).toHaveBeenCalledTimes(2);
    });
});
