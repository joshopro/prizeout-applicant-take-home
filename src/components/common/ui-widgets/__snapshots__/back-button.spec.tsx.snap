// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Back Button Component Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Move back"
    class="button back-button"
    data-testid="button"
  >
    <i
      class="far fa-chevron-left"
    />
  </button>
</DocumentFragment>
`;
